src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Types/TypeTable.cs

```diff
@@ -68,24 +68,20 @@ public TypeTable()
             _pTypeParameterTable = new Dictionary<TypeParameterSymbol, TypeParameterType>();
         }
 
-        public AggregateType LookupAggregate(Name pName, AggregateSymbol pAggregate)
+        private static KeyPair<TKey1, TKey2> MakeKey<TKey1, TKey2>(TKey1 key1, TKey2 key2) =>
+            new KeyPair<TKey1, TKey2>(key1, key2);
+
+        public AggregateType LookupAggregate(AggregateSymbol aggregate, AggregateType outer, TypeArray args)
         {
-            var key = new KeyPair<AggregateSymbol, Name>(pAggregate, pName);
-            AggregateType result;
-            if (_pAggregateTable.TryGetValue(key, out result))
-            {
-                return result;
-            }
-            return null;
+            _aggregateTable.TryGetValue(MakeKey(aggregate, MakeKey(outer, args)), out AggregateType result);
```

Comment:
Since `TryGetValue` guarantees to return the default when it fails to find something, and that's what we want to return in such a case, there's no need to check the return value.